### PR TITLE
Harvest: quick refactor for a better translation mechanism

### PR DESCRIPTION
**Pull request by gregorylegarec**
This PR uses `withLocales` from Cozy-UI.

Default exports from components are now directly translated.

It will allow us to export HOC wrapping component with translated component.

### CODE REVIEW

File: packages/cozy-harvest-lib/src/index.js [hunk 1]
```diff
@@ -1,39 +1,10 @@
...
-
-import TranslatedDeleteAccountButton from './components/DeleteAccountButton'
-import TranslatedTriggerManager from './components/TriggerManager'
-import TranslatedTriggerLauncher from './components/TriggerLauncher'
+export {
+  default as DeleteAccountButton
+} from './components/DeleteAccountButton'
+export {
+  default as withKonnectorModal
+} from './components/hoc/withKonnectorModal'
```
Comment (CPatchane) [inline packages/cozy-harvest-lib/src/index.js]: `export withKonnectorModal from './components/hoc/withKonnectorModal'` doesn't work here?

**Comment by gregorylegarec** [comment]
@ptbrowne @Crash-- @CPatchane The original implementation was causing the fr locales to not be loaded in :
* Dev environment without having pull them from Transifex
* Travis tests

As I don't wanted to log a warning every time a component would not have been able to load the fr language, and I did not want neither to catch an error silently, I came with a new implementation in the last commit, a little more complex. As it retrieve the lang from context, it require only the locales we need. What do yout think ? Do you see any simpler implementation ?

File: packages/cozy-harvest-lib/src/components/hoc/withLocales.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import withLocales from 'cozy-ui/transpiled/react/I18n/withLocales'
+
+const locales = {
+  en: require(`../../locales/en.json`),
+  fr: require(`../../locales/fr.json`)
+}
+
+export default withLocales(locales)
```
Comment (Crash--) [inline packages/cozy-harvest-lib/src/components/hoc/withLocales.js]: do you still need this file ? When you do `import withLocales from './hoc/withLocales'` is it the jsx or the js ?

Comment (gregorylegarec) [reply packages/cozy-harvest-lib/src/components/hoc/withLocales.js]: No we don't need it anymore, it was supposed to be removed in the last commit, please ignore it :)

**Comment by ptbrowne** [comment]
I think you should use a webpack context.

```
const localeContextAsObject = ctx => {
  const res = {}
  for (const filename of ctx.keys()) {
    const key = filename.slice(2, -5)
    if (key.length > 0) {
        res[key] = ctx(filename)
    }
  }
  return res
}

const locales = localeContextAsObject(require.context('./locales', false, '\.json$'))

export withLocales(locales)

**Comment by gregorylegarec** [comment]
Now waiting for https://github.com/cozy/cozy-ui/pull/1019

File: packages/cozy-harvest-lib/src/index.js (hunk 1 above)
Comment (gregorylegarec) [reply packages/cozy-harvest-lib/src/index.js]: When using this syntax, I got `Support for the experimental syntax 'exportDefaultFrom' isn't currently enabled` in Cozy-Bar :sob:

**Pull request merged**